core/src/main/java/org/apache/calcite/sql/dialect/SnowflakeSqlDialect.java

```diff
@@ -288,11 +316,118 @@ private void unparseOtherFunction(SqlWriter writer, SqlCall call, int leftPrec,
       DateTimestampFormatUtil dateTimestampFormatUtil = new DateTimestampFormatUtil();
       dateTimestampFormatUtil.unparseCall(writer, call, leftPrec, rightPrec);
       break;
+    case "PARSE_DATE":
+      unparseToDate(writer, call, leftPrec, rightPrec);
+      break;
+    case "TIME_SUB":
+      unparseTimeSub(writer, call, leftPrec, rightPrec);
+      break;
+    case "TO_HEX":
+      unparseToHex(writer, call, leftPrec, rightPrec);
+      break;
+    case "REGEXP_CONTAINS":
+      unparseRegexContains(writer, call, leftPrec, rightPrec);
+      break;
     default:
       super.unparseCall(writer, call, leftPrec, rightPrec);
     }
   }
 
+  private void unparseRegexContains(SqlWriter writer, SqlCall call, int leftPrec,
+      int rightPrec) {
+    final SqlWriter.Frame regexpLikeFrame = writer.startFunCall("REGEXP_LIKE");
+    for (SqlNode operand : call.getOperandList()) {
+      writer.sep(",");
+      operand.unparse(writer, leftPrec, rightPrec);
+    }
+    writer.endFunCall(regexpLikeFrame);
+  }
+
+  private void unparseToHex(SqlWriter writer, SqlCall call, int leftPrec, int rightPrec) {
+    SqlNode[] operands = new SqlNode[] {
+        call.operand(0), SqlLiteral.createCharString("UTF-8", SqlParserPos.ZERO)
+    };
+    SqlBasicCall toBinaryCall = new SqlBasicCall(SqlLibraryOperators.TO_BINARY, operands,
+        SqlParserPos.ZERO);
+    SqlNode varcharSqlCall =
+        getCastSpec(new BasicSqlType(RelDataTypeSystem.DEFAULT, SqlTypeName.VARCHAR, 100));
+    SqlCall castCall = CAST.createCall(SqlParserPos.ZERO, toBinaryCall, varcharSqlCall);
+    castCall.unparse(writer, leftPrec, rightPrec);
+  }
+
+  private void unparseTimeSub(SqlWriter writer, SqlCall call, int leftPrec, int rightPrec) {
+    final SqlWriter.Frame timeAddFrame = writer.startFunCall("TIMEADD");
+    SqlBasicCall operand1 = call.operand(1);
+    writer.print(operand1.getOperator().getName().replace("INTERVAL_", "") + "S");
```

Comment:
Can we do the same thing in a different way instead of a string operation

Reply:
done